### ISSUE TITLE
Rule: image-has-name changes

### ISSUE THREAD

**Issue by jkodu** [dequelabs/axe-core#1141]
The following should fail:

Non-image element with image role but no accessible name.
```html
<div role="image"></div>
```

**Comment by WilcoFiers** [comment]
This test case is incorrect. Should be `<div role="img"></div>`.
